src/main.rs

```diff
@@ -191,5 +268,26 @@ async fn basic_test() {
 
     // Start the backend and wait for SIGINT signal.
     d_web_backend.start().await.expect("Unable to start");
+    d_web_backend.create_group().await.expect("Unable to create group");
+
+    // Set and get group name
+    let group_key = d_web_backend.groups.keys().next().cloned().expect(GROUP_NOT_FOUND);
+    let group = d_web_backend.get_group(group_key.clone()).await.expect(GROUP_NOT_FOUND);
+    group.set_name(TEST_GROUP_NAME).await.expect(UNABLE_TO_SET_GROUP_NAME);
+    let name = group.get_name().await.expect(UNABLE_TO_GET_GROUP_NAME);
+    assert_eq!(name, TEST_GROUP_NAME);
+
+    // Store the group's keypair in the protected store
+    let protected_store = d_web_backend.veilid_api.as_ref().unwrap().protected_store().unwrap();
+    group.store_keypair(&protected_store).await.expect(UNABLE_TO_STORE_KEYPAIR);
+
+    // Retrieve the group's keypair from the protected store
+    let keypair_data = protected_store.load_user_secret(group_key.to_string()).await.expect(FAILED_TO_LOAD_KEYPAIR).expect(KEYPAIR_NOT_FOUND);
```

Comment:
Instead we should stop the backend, then start it again and check that we can load the group from the public key